tools/PHP_CLI_Helper.class.inc

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Use bash to determine if certain unix tools are installed.  Which is used for
+ * generic system tools (e.g. wget), and dpkg if which fails.
+ *
+ * @param string $tool Name of tool
+ *
+ * @return bool representing if tool is installed
+ */
+function installed($tool) : bool
+{
+    // `which` returns empty if a tool is not installed.
+    // shell_exec captures this output.
+    if (shell_exec("which $tool")) {
```

Comment:
I think this would be more reliable if it used exec and checked the shell return code, rather than trying to parse stdout.